sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConsumer.cs

```diff
@@ -147,114 +147,86 @@ private void CloseLink(ReceivingAmqpLink link)
-                        // return them.
-
-                        if ((messagesReceived) && (amqpMessages != null))
-                        {
-                            receivedMessages ??= new List<ServiceBusMessage>();
-
-                            foreach (AmqpMessage message in amqpMessages)
-                            {
-                                //link.DisposeDelivery(message, true, AmqpConstants.AcceptedOutcome);
-                                receivedMessages.Add(AmqpMessageConverter.AmqpMessageToSBMessage(message));
-                                message.Dispose();
-                            }
-
-                            receivedMessageCount = receivedMessages.Count;
-                            return receivedMessages;
-                        }
-
-                        // No events were available.
-
-                        return Enumerable.Empty<ServiceBusMessage>();
-                    }
-                    catch (ServiceBusException ex) when (ex.Reason == ServiceBusException.FailureReason.ServiceTimeout)
-                    {
-                        // Because the timeout specified with the request is intended to be the maximum
-                        // amount of time to wait for events, a timeout isn't considered an error condition,
-                        // rather a sign that no events were available in the requested period.
-
-                        return Enumerable.Empty<ServiceBusMessage>();
-                    }
-                    catch (Exception ex)
-                    {
-                        Exception activeEx = ex.TranslateServiceException(EntityName);
-
-                        // Determine if there should be a retry for the next attempt; if so enforce the delay but do not quit the loop.
-                        // Otherwise, bubble the exception.
-
-                        ++failedAttemptCount;
-                        retryDelay = _retryPolicy.CalculateRetryDelay(activeEx, failedAttemptCount);
-
-                        if ((retryDelay.HasValue) && (!ConnectionScope.IsDisposed) && (!cancellationToken.IsCancellationRequested))
-                        {
-                            ServiceBusEventSource.Log.MessageReceiveError(EntityName, activeEx.Message);
-                            await Task.Delay(UseMinimum(retryDelay.Value, waitTime.CalculateRemaining(stopWatch.Elapsed)), cancellationToken).ConfigureAwait(false);
-
-                            tryTimeout = _retryPolicy.CalculateTryTimeout(failedAttemptCount);
-                        }
-                        else if (ex is AmqpException)
-                        {
-                            throw activeEx;
-                        }
-                        else
-                        {
-                            throw;
-                        }
-                    }
+                    //link.DisposeDelivery(message, true, AmqpConstants.AcceptedOutcome);
```

Comment:
Do we need this commented out code when in ReceiveAndDelete mode? 
It looks like Track 1 does this.

Reply:
Uncommitted this code!

Reply:
Uncommented this code!

Reply:
Uncommented this code; although I'm not sure how it gets used in ReceiveAndDelete.